php-classes/Emergence/People/InvitationsRequestHandler.php

```diff
@@ -0,0 +1,238 @@
+    ];
+
+    public static function handleRequest()
+    {
+        switch (static::shiftPath()) {
+            case 'variables':
+                return static::handleVariablesRequest();
+            case 'preview':
+                return static::handlePreviewRequest();
+            case 'send':
+                return static::handleSendRequest();
+            case 'accept':
+                return static::handleAcceptRequest();
+            default:
+                return static::throwInvalidRequestError();
+        }
+    }
+
+    public static function handleVariablesRequest()
+    {
+        $GLOBALS['Session']->requireAccountLevel('Staff');
+
+        if (empty($_REQUEST['personId']) || !($Recipient = Person::getByID($_REQUEST['personId']))) {
+            return static::throwInvalidRequestError('personId required');
+        }
+
+        // create phantom invitation
+        $Invitation = Invitation::create([
+            'Recipient' => $Recipient
+        ]);
+
+        return static::respond('preview', [
+            'success' => true,
+            'data' => static::_getMessageVariables($Invitation)
+        ]);
+    }
+
+    public static function handlePreviewRequest()
+    {
+        $GLOBALS['Session']->requireAccountLevel('Staff');
+
+        if (empty($_REQUEST['personId']) || !($Recipient = Person::getByID($_REQUEST['personId']))) {
+            return static::throwInvalidRequestError('personId required');
+        }
+
+        // create phantom invitation
+        $Invitation = Invitation::create([
+            'Recipient' =>$Recipient
+        ]);
+
+        $emailData = Mailer::renderTemplate('loginInvitation', [
+            'Sender' => $GLOBALS['Session']->Person,
+            'Invitation' => $Invitation,
+            'message' => static::_getMessageFromTemplate($_REQUEST['message'], $Invitation)
+        ]);
+
+        $emailData['to'] = $Recipient->EmailRecipient;
+
+        return static::respond('preview', $emailData);
+    }
+
+    public static function handleSendRequest()
+    {
+        $GLOBALS['Session']->requireAccountLevel('Staff');
+
+        if (empty($_POST['people']) || !is_array($_POST['people'])) {
+            return static::throwInvalidRequestError('people required');
+        }
+
+        // pre-flight loop over people to ensure they're all valid
+        $people = [];
+        foreach($_POST['people'] AS $personID) {
+            if (!is_numeric($personID) || !($Person = Person::getByID($personID))) {
+                return static::throwInvalidRequestError('One or more of the requested invitation recipients was not found in the database.');
+            }
+
+            if (!$Person->Email) {
+                return static::throwInvalidRequestError('One or more of the requested invitation recipients does not have an email address on record.');
+            }
+
+            $people[] = $Person;
+        }
+
+        // create and send invitations
+        foreach($people AS $Person) {
```

Comment:
@nbey styling issue here and other parts of this file -- there should be a space between `foreach` and `(`